### PR TITLE
[corechecks/snmp] Add profile metadata match syntax

### PR DESCRIPTION
**Pull request by AlexandreYang**
### What does this PR do?

[corechecks/snmp] Add profile metadata match syntax

### Motivation

Needed for cases where multiple regexes are needed to handle all cases. See example usage here:
https://github.com/DataDog/integrations-core/pull/10767

```yaml
metadata:
  device:
    fields:
      os_name:
        # TODO: Add tests (possibly unit tests) for profile regex patterns
        #       Right now, we can only e2e test profiles but for match (match_pattern/match_value) and extract_value,
        #       we need a way to unit test profiles. Using e2e is much heavier and is not suitable to testing regex patterns.
        #       Some changes to test tooling might be needed to make easy to test those cases.
        symbols:
          - OID: 1.3.6.1.2.1.1.1.0
            name: sysDescr
            match_pattern: 'Cisco Internetwork Operating System Software'
            match_value: 'IOS'
            # Cisco Internetwork Operating System Software ..IOS (tm) C2600 Software (C2600-I-M), Version 12.0(3)T3,  RELEASE SOFTWARE (fc1)..Copyright (c) 1986-1999 by cisco Systems, Inc...
          - OID: 1.3.6.1.2.1.1.1.0
            name: sysDescr
            match_pattern: 'Cisco IOS Software'
            match_value: 'IOS'
            # Example: Cisco IOS Software, IOS-XE Software, Catalyst L3 Switch Software (CAT3K_CAA-UNIVERSALK9-M), Version 03.06.06E RELEASE SOFTWARE (fc1) Technical Support: http://www.cisco.com/techsupport Copyright (c) 1986-2016 by Cisco Systems, Inc.
          - OID: 1.3.6.1.2.1.1.1.0
            name: sysDescr
            match_pattern: 'Cisco NX-OS'
            match_value: 'NXOS'
            # Example: Cisco NX-OS(tm) m9100, Software (m9100-s2ek9-mz), Version 4.1(1c), RELEASE SOFTWARE Copyright (c) 2002-2008 by Cisco Systems, Inc. Compiled 11/24/2008 18:00:00
          - OID: 1.3.6.1.2.1.1.1.0
            name: sysDescr
            match_pattern: 'Cisco IOS XR'
            match_value: 'IOSXR'
            # Example: Cisco IOS XR Software (Cisco ASR9K Series), Version 4.2.3[Default] Copyright (c) 2013 by Cisco Systems, Inc.

```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/collector/corechecks/snmp/report/report_utils.go [hunk 1]
```diff
@@ -29,15 +22,40 @@ func getColumnValueFromSymbol(values *valuestore.ResultValueStore, symbol checkc
...
+		strValue, err := value.ToString()
+		if err != nil {
+			log.Debugf("error converting value to string (value=%v):", value, err)
+			return valuestore.ResultValue{}, err
+		}
+
+		if symbol.MatchPatternCompiled.MatchString(strValue) {
+			replacedVal := checkconfig.RegexReplaceValue(strValue, symbol.MatchPatternCompiled, symbol.MatchValue)
+			if replacedVal == "" {
+				return valuestore.ResultValue{}, fmt.Errorf("pattern `%v` failed to match `%v` with template `%s`", symbol.MatchPattern, strValue, symbol.MatchValue)
```
Comment (FlorianVeaux) [inline pkg/collector/corechecks/snmp/report/report_utils.go:53]: If we're here it means that the pattern matched but that the template is not compatible with the regex pattern (wrong number of groups likely?).
Could we maybe have an error message that reflects that?

File: pkg/collector/corechecks/snmp/report/report_utils.go [hunk 2]
```diff
@@ -29,15 +22,40 @@ func getColumnValueFromSymbol(values *valuestore.ResultValueStore, symbol checkc
...
+		if err != nil {
+			continue
 		}
-		newValues[index] = value
+		newValues[index] = newValue
 	}
 	return newValues, nil
 }
+
+func processValue(symbol checkconfig.SymbolConfig, value valuestore.ResultValue) (valuestore.ResultValue, error) {
```
Comment (FlorianVeaux) [inline pkg/collector/corechecks/snmp/report/report_utils.go:34]: Maybe the name could be more self-explanatory.

**Review by FlorianVeaux — APPROVED** [review]
Just a couple of naming nits but the logic looks great 👍

File: pkg/collector/corechecks/snmp/report/report_utils.go (hunk 1 above)
Comment (AlexandreYang) [reply pkg/collector/corechecks/snmp/report/report_utils.go]: Indeed, the message could be better. Update the message here: 2f3198bd5

File: pkg/collector/corechecks/snmp/report/report_utils.go (hunk 2 above)
Comment (AlexandreYang) [reply pkg/collector/corechecks/snmp/report/report_utils.go]: Indeed. Update to `processValueUsingSymbolConfig` here 2f3198bd5

**Pull request merged**